File: DSA_python/DSA08001 CẤU TRÚC DỮ LIỆU HÀNG ĐỢI 1.py
def submit():
    n = int(input())
    A = list()

    for i in range(n):
        s = input().split()

        if s[0] == '1':
            print(len(A))
        elif s[0] == '2':
            print("YES" if len(A) == 0 else "NO")
        elif s[0] == '3':
            A.append(s[1])
        elif s[0] == '4':
            if len(A) != 0: A.remove(A[0])
        elif s[0] == '5':
            print(-1 if len(A) == 0 else A[0])
        elif s[0] == '6':
            print(-1 if len(A) == 0 else A[len(A)-1])


test = int(input())
for i in range(test):
    submit()

File: DSA/DSA08001 CẤU TRÚC DỮ LIỆU HÀNG ĐỢI 1.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    deque<int> A;

    for (int i = 0; i < n; i++)
    {
        int f, s;
        cin >> f;
        if (f == 3) cin >> s;

        switch (f)
        {
            case 1:
                cout << A.size() << endl;
                break;
            case 2:
                A.empty() ? cout << "YES" : cout << "NO";
                cout << endl;
                break;
            case 3:
                A.push_back(s);
                break;
            case 4:
                if (!A.empty()) A.pop_front();
                break;
            case 5:
                A.empty() ? cout << -1 : cout << A.front();
                cout << endl;
                break;
            case 6:
                A.empty() ? cout << -1 : cout << A.back();
                cout << endl;
                break;
        }
    }
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed down input/output
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();
    
    while (so_bo_test--)
    {
        submit();
    }

    return 0;
}